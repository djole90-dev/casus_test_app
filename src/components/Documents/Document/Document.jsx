import React from 'react';
import { DocumentBody, Title } from './Document.styles';
import C from '../C_Logo/C_Logo'

const Document = ({ title, category }) => (
  <DocumentBody>
    <Title>{title}</Title>
    <C category={category}/>
  </DocumentBody>
);

export default Document;
