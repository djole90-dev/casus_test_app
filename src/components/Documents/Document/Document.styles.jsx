import styled from 'styled-components';
import { Box, Typography } from '@material-ui/core';

export const DocumentBody = styled(Box)`
  width:192px;
  height: 240px;
  padding: 32px 12px 12px 24px;
  display:flex;
  flex-direction: column;
  background: #ffffff;
  margin-bottom:39px;
  margin-left:32px;
  cursor: pointer;
  position:relative;
`;
export const Title = styled(Typography)`
  font-size:15px;
  line-height: 1.33;
  letter-spacing: 0.21px;
  color: ${({ theme }) => theme.palette.primary.main};
  font-weight:500;
`;

export const C = styled('img')`
  height:18px;
  width: 16px;
  position: absolute;
  left: 14px;
  bottom: 12px;
`;
