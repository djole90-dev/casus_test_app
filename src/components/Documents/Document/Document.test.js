import React from 'react';
import { shallow } from 'enzyme';
import Document from './Document';
import C from '../C_Logo/C_Logo';

import { Title, DocumentBody } from './Document.styles';

const setUp = (Component, props = {}) => {
  const component = shallow(<Component {...props} />);
  return component;
};

describe('Document Component', () => {
  const baseProps = {
    title: 'Title',
    category: '123'
  };

  it('should match snapshot', () => {
    const component = setUp(Document, baseProps);
    expect(component).toMatchSnapshot();
  });

  it('should render properly', () => {
    const component = setUp(Document, baseProps);
    const docwrapper = component.find(DocumentBody)
    const title = docwrapper.children().find(Title)
    const logo = docwrapper.children().find(C)
    expect(docwrapper.length).toBe(1);
    expect(title.length).toBe(1)
    expect(title.text()).toEqual('Title');
    expect(logo.length).toBe(1);
  });
});
