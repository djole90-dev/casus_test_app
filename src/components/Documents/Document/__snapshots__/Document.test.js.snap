// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document Component should match snapshot 1`] = `
<Styled(Styled(MuiBox))>
  <Styled(WithStyles(ForwardRef(Typography)))>
    Title
  </Styled(WithStyles(ForwardRef(Typography)))>
  <C_Logo
    category="123"
  />
</Styled(Styled(MuiBox))>
`;
